### PR TITLE
OLS-694: Improve memory usage in collector script by lazily loading llama_index

### PR DESCRIPTION
**Pull request by onmete**
## Description

With this, the memory consumption for the sidecar falls from 500MiB under 100MiB.

## Type of change

- [x] Optimization

### CODE REVIEW

**Comment by bparees** [comment]
/approve

/assign @tisnik 

@tisnik probably best for you to assess this from a python practices perspective, but the direction makes sense to me.

File: Makefile [hunk 1]
```diff
@@ -65,7 +65,7 @@ coverage-report:	test-unit ## Export unit test coverage report into interactive
 	coverage html --data-file="${ARTIFACT_DIR}/.coverage.unit"
 
 check-types: ## Checks type hints in sources
-	mypy --explicit-package-bases --disallow-untyped-calls --disallow-untyped-defs --disallow-incomplete-defs ols/
+	mypy --explicit-package-bases --disallow-untyped-calls --disallow-untyped-defs --disallow-incomplete-defs --config-file pyproject.toml ols/
```
Comment (tisnik) [inline Makefile:68]: this should not be needed, really

File: ols/src/rag_index/index_loader.py [hunk 2]
```diff
@@ -88,7 +95,7 @@ def _load_index(self) -> None:
                 logger.exception(f"Error loading vector index:\n{err}")
 
     @property
-    def vector_index(self) -> Optional[BaseIndex]:
+    def vector_index(self):
```
Comment (tisnik) [inline ols/src/rag_index/index_loader.py:98]: not sure if it's good approach tbh, but ok (it will kick us later)

**Review by tisnik — COMMENTED** [review]
the idea is correct, just having some doubts (as usual)

**Pull request merged**